src/main/scala/com/databricks/spark/sql/perf/query.scala

```diff
@@ -59,11 +72,17 @@ case class QueryForTest(
         Seq.empty[BreakdownResult]
       }
 
-      // The executionTime for the entire query includes the time of type conversion from catalyst to scala.
-      val executionTime = if (query.collectResults) {
-        benchmarkMs { dataFrame.rdd.collect() }
-      } else {
-        benchmarkMs { dataFrame.rdd.foreach {row => Unit } }
+      // The executionTime for the entire query includes the time of type conversion
+      // from catalyst to scala.
+      val executionTime = benchmarkMs {
+        query.executionMode match {
+          case CollectResults => dataFrame.rdd.collect()
+          case ForeachResults => dataFrame.rdd.foreach { row => Unit }
+          case WriteParquet(location) => {
+            dataFrame.rdd.collect()
+            dataFrame.saveAsParquetFile(s"$location/$name.parquet")
```

Comment:
If we do `collect` first and then do `saveAsParquetFile`, we will execute the query twice. I think we can remove `dataFrame.rdd.collect()`. Also, I guess it is alternative option of `ForeachResults` to benchmark queries having many output rows. So, I think we'd better not to call collect, which will retrieve all results to the driver.